### PR TITLE
Test database with new "jsonless" schema.

### PR DESCRIPTION
**Pull request by 0xf8f8ff**

### CODE REVIEW

File: doc/dev/testdatabase/testdb.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Testing database
+
+The database (`testing.db`) is using the updated Quickfeed schema without any json-encoded string fields, `DisableForeignKeyConstraintWhenMigrating` (`gormdb.go:48`) should not be set to true.
+
+- The easiest way to inspect or edit the database is to use a sqlite database browser GUI, for example the [DB Browser](https://sqlitebrowser.org/) or any other alternative.
+
+- User with ID 1 is a super user registered as `CourseCreator` of all courses in the database. The `RemoteIdentity` record for this user can be replaced with Github's remote ID and access token of your Github user to login as super user.
```
[the remark this answers is not in the document]
Comment (meling) [reply doc/dev/testdatabase/testdb.md:7]: Why is it called `CourseCreator` for all courses? I would probably say that ID 1 is the admin user with access to all courses in the database and access to the GitHub quickfeed user -- in the current deployment, this is currently the `autograder-uis` GitHub user.

File: doc/dev/testdatabase/testdb.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+- Course with ID 1 is based on the [qf406](https://github.com/qf406) created specifically for testing purposes. This repository can be used for any kind of testing: adding or removing users and groups/teams, adding assignments and tests, pushing submissions to student repos and so on. Request an invite to the repository if you want to use it for testing.
```
Comment (meling) [inline doc/dev/testdatabase/testdb.md:9]: ... on the qf406 *organization* created ... This *organization*...

**Review by meling — CHANGES_REQUESTED** [review]

File: doc/dev/testdatabase/testdb.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xf8f8ff) [reply doc/dev/testdatabase/testdb.md:7]: I mean, each course has a `CourseCreator` user ID reference and some course related methods can only be called by this user. In the current deployment it is never the `autograder-uis` user as far as I can see. But I can of course remove it if you think it is not worth mentioning.

File: doc/dev/testdatabase/testdb.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Testing database
```
[the remark this answers is not in the document]
Comment (0xf8f8ff) [reply doc/dev/testdatabase/testdb.md]: Done.

File: doc/dev/testdatabase/testdb.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+The database (`testing.db`) is using the updated Quickfeed schema without any json-encoded string fields, `DisableForeignKeyConstraintWhenMigrating` (`gormdb.go:48`) should not be set to true.
```
[the remark this answers is not in the document]
Comment (0xf8f8ff) [reply doc/dev/testdatabase/testdb.md]: Done.

File: doc/dev/testdatabase/testdb.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+- The easiest way to inspect or edit the database is to use a sqlite database browser GUI, for example the [DB Browser](https://sqlitebrowser.org/) or any other alternative.
```
[the remark this answers is not in the document]
Comment (0xf8f8ff) [reply doc/dev/testdatabase/testdb.md]: Added.

File: doc/dev/testdatabase/testdb.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xf8f8ff) [reply doc/dev/testdatabase/testdb.md]: Added SQL command.

File: doc/dev/testdatabase/testdb.md (hunk 2 above)
Comment (0xf8f8ff) [reply doc/dev/testdatabase/testdb.md]: Fixed.

File: doc/dev/testdatabase/testdb.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (meling) [reply doc/dev/testdatabase/testdb.md]: > I mean, each course has a `CourseCreator` user ID reference and some course related methods can only be called by this user. In the current deployment it is never the `autograder-uis` user as far as I can see. But I can of course remove it if you think it is not worth mentioning.

Hmm. Weird. I thought we used the `autograder-uis` GitHub user name (that I control) as the first user to log on to the ag3 server, which then becomes the main admin user.

[the remark this answers is not in the document]
Comment (0xf8f8ff) [reply doc/dev/testdatabase/testdb.md]: But this is correct, `autograder-uis` is the main admin user. `CourseCreator` is a completely different role, the main teacher of each course who can update enrollments and so on.

**Review by meling — APPROVED** [review]

**Pull request merged**